### PR TITLE
[Technical design] Design for performance notification (alert) system

### PR DESCRIPTION
**Pull request by taoran6**
**Linked GitHub issue ID**: #  

## Pull Request Checklist

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Code compiles correctly with all tests are passed.
- [ ] I've read the [contributing guide](https://github.com/microsoft/HydraLab/blob/main/CONTRIBUTING.md#making-changes-to-the-code) and followed the recommended practices.
- [ ] [Wikis](https://github.com/microsoft/HydraLab/wiki) or [README](https://github.com/microsoft/HydraLab/blob/main/README.md) have been reviewed and added / updated if needed (for bug fixes / features)

### Does this introduce a breaking change?
*If this introduces a breaking change for Hydra Lab users, please describe the impact and migration path.*

- [ ] Yes
- [x] No

## Pull Request Description

**A few words to explain your changes**:  
Design for performance notification (alert) system

### How you tested it
*Please make sure the change is tested, you can test it by adding UTs, do local test and share the screenshots, etc.*

Please check the type of change your PR introduces:
- [ ] Bugfix
- [ ] Feature
- [x] Technical design
- [ ] Build related changes
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Code style update (formatting, renaming) or Documentation content changes
- [ ] Other (please describe): 

### Feature UI screenshots or Technical design diagrams
*If this is a relatively large or complex change, kick it off by drawing the tech design with PlantUML and explaining why you chose the solution you did and what alternatives you considered, etc...*
image

Class diagram:
image

### Reference
[ HTTP Trigger | A Power Automate Tutorial](https://blog.enterprisedna.co/microsoft-flow-http-trigger-a-power-automate-tutorial/)
[微信支付通知API](https://pay.weixin.qq.com/wiki/doc/apiv3/apis/chapter3_1_5.shtml)

### CODE REVIEW

File: agent/doc/UML/performance_test_design.puml [hunk 1]
```diff
@@ -164,3 +164,45 @@ note right
 end note
 @enduml
 
+@startuml performance_test_notify_uml
+class TestTask {
+    String perfNotifyUrl
+}
+
+class PerformanceNotification {
```
Comment (taoran6) [inline agent/doc/UML/performance_test_design.puml:172]:
```suggestion
class TestNotification {
```

File: agent/doc/UML/performance_test_design.puml [hunk 2]
```diff
@@ -164,3 +164,45 @@ note right
...
+    String pipelineLink;
+    String testTaskId;
+    List<PerformanceTestResult> resultList;
```
Comment (taoran6) [inline agent/doc/UML/performance_test_design.puml:175]:
```suggestion
    Object content;
```

File: agent/doc/UML/performance_test_design.puml [hunk 3]
```diff
@@ -164,3 +164,45 @@ note right
...
+    long testStartTimeMillis;
+}
+
+class PerformanceTestNotifier {
+    + void sendPerformanceNotification(List<PerformanceTestResult> resultList, TestTask testTask, Logger logger)
+}
+
+PerformanceTestManagementService -u-> TestTask
+PerformanceTestNotifier o- PerformanceTestManagementService
```
Comment (taoran6) [inline agent/doc/UML/performance_test_design.puml:184]: Reflect the relation with PerformanceTestListener.

File: agent/doc/UML/performance_test_design.puml [hunk 4]
```diff
@@ -164,3 +164,45 @@ note right
 end note
 @enduml
 
+@startuml performance_test_notify_uml
+class TestTask {
+    String perfNotifyUrl
```
Comment (taoran6) [inline agent/doc/UML/performance_test_design.puml:169]:
```suggestion
    String notificationURL
```

File: agent/doc/UML/performance_test_design.puml [hunk 5]
```diff
@@ -164,3 +164,45 @@ note right
...
+
+class PerformanceNotification {
+    String pipelineLink;
+    String testTaskId;
+    List<PerformanceTestResult> resultList;
+    long testStartTimeMillis;
+}
+
+class PerformanceTestNotifier {
+    + void sendPerformanceNotification(List<PerformanceTestResult> resultList, TestTask testTask, Logger logger)
```
Comment (taoran6) [inline agent/doc/UML/performance_test_design.puml:180]: encapsulate this with TestNotification

File: agent/doc/UML/performance_test_design.puml [hunk 6]
```diff
@@ -164,3 +164,45 @@ note right
...
+@startuml performance_test_notify_component
+skinparam componentStyle rectangle
+package "Hydra Lab agent" {
+    component "Performance Management" {
+        [PerformanceTestNotifier]
+        [PerformanceTestManagementService]
+    }
+}
+
+cloud "Notify Service" as Notify {
```
Comment (taoran6) [inline agent/doc/UML/performance_test_design.puml:197]:
```suggestion
cloud "3rd Party Notify Service" as Notify {
```

**Review by DexterDreeeam — APPROVED** [review]

**Review by zhou9584 — APPROVED** [review]

**Review by AC-Phoenix — APPROVED** [review]

**Pull request merged**